js/stdlib/ts/ALGO.ts

```diff
@@ -2341,21 +2343,36 @@ const balanceOfM = async (acc: Account, token: Token | null): Promise<BigNumber
         return false;
       }
     } else {
-      // Indexer token balance
-      const tokenId = bigNumberToNumber(token);
-      const client = await getAlgodClient();
-      const query = client.accountAssetInformation(addr, tokenId) as ApiCall<AccountAssetInfo>;
-      const accountAssetInfoM = await doQueryM_(dhead, query);
-      if ('val' in accountAssetInfoM) {
-        const assetHolding = accountAssetInfoM['val']['asset-holding'];
-        return assetHolding ? bigNumberify(assetHolding['amount']) : false;
-      } else {
-        return false;
-      }
+      return indexerBalanceOfMNonNet(addr, token, dhead);
     }
   }
 }
 
+const clientBalanceOfMNonNet = async (addr:string, token:Token, dhead:string) => {
+  const client = await getAlgodClient();
+  const query = client.accountAssetInformation(addr, bigNumberToNumber(token)) as unknown as ApiCall<AccountAssetInfo>;
+  const accountAssetInfoM = await doQueryM_(dhead, query);
+  if ('val' in accountAssetInfoM) {
+    return bigNumberify(accountAssetInfoM['val']['asset-holding']?.['amount'] ?? 0);
+  } else {
+    return false;
+  }
+}
+
+const indexerBalanceOfMNonNet = async (addr:string, token:Token, dhead:string) => {
```

Comment:
These functions are the same and you should just do one

Reply:
Yeah, I hadn't looked closely at what they were doing at that point, I was just taking things apart structurally.